### PR TITLE
add deployed-spark-versions.txt

### PR DESCRIPTION
**Pull request by cseed**
determines which spark version(s) are tested by PR build step

I changed the CI as follows:

 - there are now three source code subprojects: 0.1, mainline and PRs
 - mainline now only builds/deploys against 2.2.0
 - PRs builds against both 2.0.2 and 2.2.0, but!  if there is a `./deployed-spark-versions.txt` file, it only builds versions listed in that file (that is, immediately succeeds for versions not listed in that file).

As soon as this goes in, python3 should be able to pass.  Unless I broke something else, which is basically a 100% certainty.

### CODE REVIEW

**Comment by cseed** [comment]
FYI @tpoterba

**Comment by cseed** [comment]
Oh, and there are no more separate docs builds on PRs.

**Comment by tpoterba** [comment]
This looks good to me, waiting on Dan's review.

File: build.gradle [hunk 1]
```diff
@@ -34,7 +34,7 @@ buildScan {
 }
 
 String hailVersion = "devel"
-String[] deployedSparkVersions = ["2.0.2"]
+String[] deployedSparkVersions = new File("deployed-spark-versions.txt")
```
Comment (danking) [inline build.gradle]: neat, I didn't know Gradle auto-converts a file to an array of strings

**Pull request merged**